src/plugins/FileService/components/ProgressBar.tsx

```diff
@@ -0,0 +1,102 @@
+import { makeStyles, Typography } from '@material-ui/core'
+import React from 'react'
+import { useI18N } from '../../../utils/i18n-next-ui'
+import { formatFileSize } from '../utils'
+
+const useStyles = makeStyles({
+    container: {
+        height: 250,
+        flexDirection: 'column',
+        justifyContent: 'space-between',
+        alignItems: 'center',
+        userSelect: 'none',
+        paddingTop: 18,
+        paddingBottom: 18,
+    },
+    progress: {
+        display: 'flex',
+        alignItems: 'center',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        width: 400,
+    },
+    meta: {
+        display: 'flex',
+        justifyContent: 'space-between',
+        width: '100%',
+        margin: 0,
+        color: '#939393',
+        fontSize: 12,
+        lineHeight: 1.75,
+    },
+    bar: {
+        display: 'flex',
+        overflow: 'hidden',
+        width: '100%',
+        height: 5,
+        borderRadius: 6.5,
+        backgroundColor: '#f5f5f5',
+        fontSize: 12,
+        lineHeight: 0,
+    },
+    insideBar: {
+        display: 'flex',
+        overflow: 'hidden',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        transition: 'width 100ms linear',
+        textAlign: 'center',
+        whiteSpace: 'nowrap',
+        color: '#fff',
+        borderRadius: 6.5,
+        backgroundColor: '#2ca4ef',
+    },
+})
+
+interface Props {
+    preparing: boolean
+
+    startedAt: number
+    fileSize: number
+    sendSize: number
+}
+
+export const ProgressBar: React.FC<Props> = (props) => {
+    const { t } = useI18N()
+    const classes = useStyles()
+    const { startedAt, fileSize, sendSize } = props
+    const width = (sendSize / fileSize) * 100
+    const elapsed = (Date.now() - startedAt) / 1000
+    const remaining = (fileSize - sendSize) / (elapsed ? sendSize / elapsed : 0)
+    let completion = t('plugin_file_service_uploading_preparing')
+    if (!props.preparing) {
+        completion = `${formatFileSize(sendSize)} of ${formatFileSize(fileSize)}`
+    }
+    return (
+        <section className={classes.progress}>
+            <Typography className={classes.meta}>
+                <Duration value={remaining} />
+                <span>{completion}</span>
+            </Typography>
+            <p className={classes.bar}>
```

Comment:
Please don't re-invent the wheel. https://material-ui.com/components/progress/#linear-with-label